opentelemetry-collector-contrib: 0.87.0 -> 0.96.0

bouk opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bouk committed:
opentelemetry-collector-contrib: 0.87.0 -> 0.96.0

Skip some extra failing tests

uri-canva reviewed (dismissed):
Some new tests are failing, looking at the failure they might have to be run separately or skipped entirely.

bouk commented:
I've bumped it to version 0.93.0 which just came out and added skips for the failing tests @uri-canva

spacekitteh commented:
Can this be updated to 0.94?

spacekitteh commented:
I feel like the opentelemetry-collector/opentelemetry-collector-contrib/opentelemetry-cpp-contrib packages really should be on an autoupdate script.

bouk commented:
@spacekitteh I've added update scripts and bumped it to the latest. @uri-canva could you take another look?

06kellyjac commented on pkgs/tools/misc/opentelemetry-collector/contrib.nix:
```diff
@@ -41,16 +42,21 @@ buildGoModule rec {
-  # This test fails on darwin for mysterious reasons.
-  checkFlags = lib.optionals stdenv.isDarwin
-    [ "-skip" "TestDefaultExtensions/memory_ballast" ];
+  checkFlags = [
+    # collectd test tries and fails to bind to port 8081
+    # mezo test tries to reach external service
+    # memory_ballast test fails on darwin for mysterious reasons.
+    "-skip=TestDefaultReceivers/collectd|TestDefaultExporters/mezmo${if stdenv.isDarwin then "|TestDefaultExtensions/memory_ballast" else ""}"
+  ];
```
```suggestion
  checkFlags = [
    # collectd test tries and fails to bind to port 8081
    # mezmo test tries to reach external service
    "-skip=TestDefaultReceivers/collectd|TestDefaultExporters/mezmo"
  ] ++ lib.optionals stdenv.isDarwin [
    # memory_ballast test fails on darwin for mysterious reasons
    "-skip=TestDefaultExtensions/memory_ballast"
  ];
```

Different formatting to be a bit more readable and fixed a spelling issue in the comments

bouk replied:
Unfortunately you can't pass the -skip flag multiple times, it overwrites previous calls.

06kellyjac commented on pkgs/tools/misc/opentelemetry-collector/contrib.nix:
```diff
@@ -41,16 +42,21 @@ buildGoModule rec {
     "-X github.com/open-telemetry/opentelemetry-collector-contrib/internal/version.Version=v${version}"
   ];
 
+  passthru.updateScript = nix-update-script { };
```
Personally I don't put these in when it's just the basic updater as r-ryantm manages just fine and using `nix-update` myself is pretty easy. I don't see a massive amount of people using the maintainer script to trigger these.

But I don't mind that much

bouk replied:
Ah TIL, I didn't know that. I guess it just wasn't creating the PRs automatically because the tests were failing

06kellyjac reviewed (commented):
Thanks.

Some comments

bouk commented:
I've upgraded to version 0.95.0.

I've removed the update script, now that the tests are fixed @r-ryantm should automatically pick up new versions

bouk commented:
@wegank  could you review this one?

wegank commented:
The 4 neutral checks don't look too good, though...

wegank commented:
Result of `nixpkgs-review pr 281809` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>opentelemetry-collector-contrib</li>
  </ul>
</details>

bouk committed:
opentelemetry-collector-contrib: disable check

As discussed in
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31678,
the tests inside cmd/otelcontrib can't be run any more.

bouk commented:
I've asked for help upstream and it seems that the tests we're running here are deprecated: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31678

I've added a `doCheck = false`, since we were only running the tests inside `cmd/otelcontribcol` anyways. I don't have time to get the other tests working, so if this is not acceptable then I'll just close this PR


I've also bumped to the latest 0.96.0

wegank commented:
> I've added a `doCheck = false`, since we were only running the tests inside `cmd/otelcontribcol` anyways. I don't have time to get the other tests working, so if this is not acceptable then I'll just close this PR

I'm fine with that.

bouk commented:
@wegank  all the checks have passed!

wegank commented:
@uri-canva Does it look good to you?
